src/common/isolatable_hardwares.cpp

```diff
@@ -668,6 +668,65 @@ std::optional<std::vector<sdbusplus::message::object_path>>
     return listOfChildsInventoryPath;
 }
 
+std::optional<sdbusplus::message::object_path>
+    IsolatableHWs::getFRUInventoryPath(
+        const std::pair<LocationCode, InstanceId>& fruDetails,
+        const inv_path_lookup_func::LookupFuncForInvPath& fruInvPathLookupFunc)
+{
+    auto inventoryPathList = getInventoryPathsByLocCode(fruDetails.first);
+    if (!inventoryPathList.has_value())
+    {
+        return std::nullopt;
+    }
+
+    if (inventoryPathList->empty())
+    {
+        // The inventory object doesn't exist for the given location code.
+        log<level::ERR>(
+            fmt::format("The inventory object does not exist for the given "
+                        "location code [{}].",
+                        fruDetails.first)
+                .c_str());
+        return std::nullopt;
+    }
+    else if (inventoryPathList->size() == 1)
+    {
+        // Only one inventory object is exist for the given location code.
+        // For example, DIMM
+        return (*inventoryPathList)[0];
+    }
+    else
+    {
+        /**
+         * More than one inventory objects are exist for the given location code
```

Comment:
This is only applicable for processor use case , for other units it is error case right?

Reply:
I like to leave this code as common and did not like to add some special checks to support only processor use cases here.

are you seeing any issues here?

Reply:
Assumption here is for more than if more than one inventory objects are exist , you alway's expect the MRU based logic ?  If you are not sure about this assumption better to error out software failure and , can fix the problem corectly during developemnt it slef.

Reply:
>Assumption here is for more than if more than one inventory objects are exist , you alway's expect the MRU based logic ? 

Not really, This function will expect FRU details (Location code and instance id, here instance-id could be any attribute and that will be taken care of by the caller with help of the defined isolatable hardware lookup function.

> better to error out software failure and , can fix the problem corectly during developemnt it self

Agree with you i.e seeing errors during development face will help us and currently, it will throw an error if the instance did not match with the retrieved inventory path.